### PR TITLE
Smart car controller

### PR DESCRIPTION
**Pull request by AlaahT**
The car can be controlled by the terminal commands using the MQTT broker to be controlled manually as well as make it drive autonomously

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

**Review by platisd — CHANGES_REQUESTED** [review]
1. Remove and gitignore the `.DS_Store`  files
2. The new file is not built on CI
3. Why are there now two Arduino sketches?
4. Why is there commented out code in the sketch? Please either remove the code or the comments.

File: .gitignore [hunk 1]
```diff
@@ -1,70 +0,0 @@
-## C++ stuff
```
Comment (platisd) [inline .gitignore:1]: You shouldn't remove this :sweat_smile:

**Comment by AlaahT** [comment]
Oh, I misinterpreted your mail 😅 I thought you meant I should remove both gitignore and .DC_Store files. I'll try to change it now

Hämta Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

File: arduino/smartcar/SmartCarController.ino [hunk 2]
```diff
@@ -0,0 +1,262 @@
+#include <vector>
```
Comment (platisd) [inline arduino/smartcar/SmartCarController.ino:1]: Why are there two arduino files?

**Comment by AlaahT** [comment]
My thought was that we keep the first one since it was the first assignment and the second one is what we should use when integrating with the app. Should we just keep one of the files?

Hämta Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by platisd** [comment]
You shouldn't have any code that's not used, so please remove any dead code and make sure all code is built/verified on CI.

**Comment by AlaahT** [comment]
Ok, I'll do that. Thank you so much for the feedback.

Hämta Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by AlaahT** [comment]
I think I've completed all the requested changes. Is there anything I need to do?

**Review by Dahlaan — COMMENTED** [review]
Looks good but please remove file gitignore, sketch_obstacle_aavoidance.ino. Did you want to add the workflow main.yml?

**Comment by LabibaKararEshaba** [comment]
Looks good.

**Review by Dahlaan — APPROVED** [review]

**Pull request merged**